### PR TITLE
impl cache hit for job from bc from build lookup

### PR DESCRIPTION
**Pull request by gabemontero**
@openshift/sig-developer-experience @mrsiano @jeremyeder  fyi

this is an optimization I devised based on a bottleneck @mrsiano uncovered during scalability testing

### CODE REVIEW

**Comment by gabemontero** [comment]
Tests passed but same intermittent failure during extended test cleanup.

```
Ran 3 of 440 Specs in 1493.966 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 437 Skipped 

Ginkgo ran 1 suite in 24m54.885326286s
Test Suite Passed

[INFO] [04:12:20+0000] Running serial tests with focus openshift pipeline build
I0404 04:12:20.363522  101079 test.go:94] Extended test version v3.10.0-alpha.0+f7fe3b5-511
[WARNING] [04:12:21+0000] No tests were selected
I0404 04:12:22.196585  102000 test.go:94] Extended test version v3.10.0-alpha.0+f7fe3b5-511
[WARNING] [04:12:23+0000] No tests were selected
[INFO] [04:12:23+0000] [CLEANUP] Beginning cleanup routines...
[INFO] [04:12:23+0000] [CLEANUP] Dumping cluster events to _output/scripts/core/artifacts/events.txt
Logged into "https://ip-172-18-10-47.ec2.internal:8443" as "system:admin" using existing credentials.

You have access to the following projects and can switch between them with 'oc project <projectname>':

  * default
    kube-public
    kube-service-catalog
    kube-system
    logging
    management-infra
    openshift
    openshift-ansible-service-broker
    openshift-infra
    openshift-node
    openshift-sdn
    openshift-template-service-broker
    openshift-web-console

Using project "default".
[INFO] [04:12:24+0000] [CLEANUP] Dumping container logs to _output/scripts/core/logs/containers
[INFO] [04:12:32+0000] [CLEANUP] Truncating log files over 200M
[INFO] [04:12:32+0000] [CLEANUP] Stopping docker containers
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace

```

File: src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java [hunk 1]
```diff
@@ -575,6 +575,13 @@ public static WorkflowJob getJobFromBuild(Build build) {
 		if (StringUtils.isEmpty(buildConfigName)) {
 			return null;
 		}
+
+		WorkflowJob job = BuildConfigToJobMap.getJobFromBuildConfigNameNamespace(buildConfigName, 
+		        build.getMetadata().getNamespace());
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java]: i guess this is why?  we don't have the uid at this point in the code so you switched the cache to use a key that's more universally available?

**Comment by bparees** [comment]
question about why the change was needed aside, the changes lgtm.

**Comment by gabemontero** [comment]
On Wed, Apr 4, 2018 at 12:58 PM, Ben Parees <notifications@github.com>
wrote:

> *@bparees* commented on this pull request.
> ------------------------------
>
> In src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMap.java
> <https://github.com/openshift/jenkins-sync-plugin/pull/218#discussion_r179168976>
> :
>
> > @@ -30,9 +33,10 @@ static synchronized void initializeBuildConfigToJobMap() {
>                  if (buildConfigProjectProperty == null) {
>                      continue;
>                  }
> -                String bcUid = buildConfigProjectProperty.getUid();
>
> curious why using the uid wasn't working as the cache key?
> ------------------------------
>
> In src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java
> <https://github.com/openshift/jenkins-sync-plugin/pull/218#discussion_r179212682>
> :
>
> > @@ -575,6 +575,13 @@ public static WorkflowJob getJobFromBuild(Build build) {
>  		if (StringUtils.isEmpty(buildConfigName)) {
>  			return null;
>  		}
> +
> +		WorkflowJob job = BuildConfigToJobMap.getJobFromBuildConfigNameNamespace(buildConfigName,
> +		        build.getMetadata().getNamespace());
>
> i guess this is why? we don't have the uid at this point in the code so
> you switched the cache to use a key that's more universally available?
>
Yep that was the reason.

**Comment by gabemontero** [comment]
thanks @bparees 

hitting the merge button given the prior test complete but ext test cleanup flake and have another PR I want to get in today after this one.

**Comment by jeremyeder** [comment]
@gabemontero shall we consider a plan to integrate these tests in our scale ci pipeline?

**Comment by gabemontero** [comment]
@jeremyeder need a couple of clarifications

a) by `these tests`, so you mean the automated regression testing that PRs against this repo kick off ?

b) by `scale ci pipeline`, do you mean the specific tests @mrsiano is currently doing so we better understand some capacity boundaries wrt pipeline strategy builds?  Or is there some acutal ci system you team has outside of the tests @mrsiano is doing?

**Pull request merged**